akka-cluster-sharding/src/main/scala/akka/cluster/sharding/Shard.scala

```diff
@@ -358,17 +358,29 @@ private[akka] class RememberEntityStarter(
   shardId:   ShardRegion.ShardId,
   ids:       Set[ShardRegion.EntityId],
   settings:  ClusterShardingSettings,
-  requestor: ActorRef
-) extends Actor {
+  requestor: ActorRef) extends Actor with ActorLogging {
 
   import context.dispatcher
   import scala.concurrent.duration._
 
   case object Tick
 
-  val region = ClusterSharding(context.system).shardRegion(typeName)
   var waitingForAck = ids
 
+  var region: Option[ActorRef] = None
+  private def initRegion(logWarning: Boolean = true): Unit = region match {
+    case s @ Some(ref) ⇒ // already initialized
+    case None ⇒
+      try {
+        region = Some(ClusterSharding(context.system).shardRegion(typeName))
+      } catch {
+        case e: IllegalArgumentException ⇒
+          if (logWarning)
+            log.warning("Sharding not initialized for []", typeName)
+      }
+  }
+
+  initRegion(logWarning = false)
```

Comment:
How can the region not be started when the shard is triggering this one to start? Isn't the shards a child of the region even?